com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -370,13 +220,51 @@ void UpgradePrefabsAndPrefabInstances()
                     conversionLinks.Add(conversionLink);
                     PrefabUtility.RecordPrefabInstancePropertyModifications(go); // record name change
                 }
-
-                conversionLinksPerScene.Add(m_SceneManager.GetScenePath(s), conversionLinks);
+                conversionLinksPerScene.Add(s, conversionLinks);
                 // save changes (i.e. converted prefab instances)
                 EditorSceneManager.SaveScene(scene); 
             }
 
-            // Upgrade Prefab Assets
+            return conversionLinksPerScene;
+        }
+        
+        static List<GameObject> GetAllGameObjects()
+        {
+            var all = (GameObject[])Resources.FindObjectsOfTypeAll(typeof(GameObject));
+            return all.Where(go => !EditorUtility.IsPersistent(go.transform.root.gameObject)
+                && (go.hideFlags & (HideFlags.NotEditable | HideFlags.HideAndDontSave)) == 0).ToList();
+        }
+
+        static void RestoreConvertedCopyReferences(GameObject original, GameObject converted, List<ConversionLink> conversionLinks)
+        {
+            // TODO: if we'll have more components that can reference upgradables, then we'll need to
+            // TODO: create some Data in UpgradeObjectToCM3.Data to make this generic and loopable
```

Comment:
I think if we have some general reflection-based code to handle references in any script, this problem will solve itself